### PR TITLE
added hw-7-1

### PR DESCRIPTION
**Pull request by kenu21**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -3,50 +3,112 @@
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List</p>
  */
+
+import java.util.Arrays;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] array = new Object[16];
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: why 16?

better rename to `elements` or `dataElements`

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -3,50 +3,112 @@
...
     public void add(T value, int index) {
-
+        if (value == null) {
+            return;
+        }
+        if (index < 0 || index > indexTopArray) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        if (indexTopArray >= array.length) {
+            array = resizeAndCopyArray(array);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: can we make method resizeAndCopyArray() void?

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -3,50 +3,112 @@
...
+        }
+        if (index == indexTopArray) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: do we need this checking?

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -3,50 +3,112 @@
...
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
+            return;
+        }
+        for (int i = 0; i < list.size(); i++) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: optional:
see how this method is implemented in the source code

File: src/main/java/core/basesyntax/ArrayList.java [hunk 5]
```diff
@@ -3,50 +3,112 @@
...
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= indexTopArray) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        T t = (T) array[index];
+        for (int i = index + 1; i < indexTopArray; i++) {
+            array[i - 1] = array[i];
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: we can use method System.arraycopy()

File: src/main/java/core/basesyntax/ArrayList.java [hunk 6]
```diff
@@ -3,50 +3,112 @@
...
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return indexTopArray < 1;
+    }
+
+    private Object[] resizeAndCopyArray(Object[] input) {
+        return Arrays.copyOf(input, input.length * 2);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: why we increment size in twice?

Comment (kenu21) [reply src/main/java/core/basesyntax/ArrayList.java]: We increment size in 1.5, for example, to don't waste time ever time when we have full array.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 3 above)
Comment (kenu21) [reply src/main/java/core/basesyntax/ArrayList.java]: No, we don't need it. I will fix.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 2 above)
Comment (kenu21) [reply src/main/java/core/basesyntax/ArrayList.java]: Yes, I will fix.

File: src/main/java/core/basesyntax/ArrayList.java (hunk 1 above)
Comment (kenu21) [reply src/main/java/core/basesyntax/ArrayList.java]: You said me at retrospective that 8 and 16 are good number for arrays :-) 
I don't have any "elements". What did you mean?

**Pull request closed without merging**